bundles/edu.kit.ipd.sdq.activextendannotations/src/edu/kit/ipd/sdq/activextendannotations/Lazy.xtend

```diff
@@ -53,50 +60,79 @@ class LazyProcessor extends AbstractFieldProcessor {
 			field.addError("A lazy field must have an initializer.")
 			return
 		}
-			
-		if (field.visibility == org.eclipse.xtend.lib.macro.declaration.Visibility.PUBLIC) { 
+
+		if (field.visibility == org.eclipse.xtend.lib.macro.declaration.Visibility.PUBLIC) {
 			field.addWarning("A lazy field should not be public, as this makes internals visible to the outside.")
 		}
-			
+
 		val setter = field.declaringType.findDeclaredMethod('set' + field.simpleName.toFirstUpper, field.type)
 		if (setter !== null) {
 			setter.addError("A lazy field cannot have a setter.")
 		}
 
+		val annotation = field.findAnnotation(Lazy.findTypeGlobally)
+		val getterVisibility = Visibility.valueOf(annotation.getEnumValue('value').simpleName).
+			toXtendVisibility(field.visibility)
+		val synchronizeAccess = annotation.getBooleanValue('synchronizeInitialization')
+
 		val isInited = field.declaringType.addField('''_«field.simpleName»_isInitialised''') [
 			type = context.primitiveBoolean
 			initializer = '''false'''
 			visibility = field.visibility
```

Comment:
Is it necessary/reasonable to make the isInitialized-flag potentially visible outside the class (depending on the field visibility)? For me it would feel natural to have that flag encapsulated as an internal state representation.

Reply:
Done, we already talked about that :-)